libs/agno/agno/vectordb/qdrant/qdrant.py

```diff
@@ -309,13 +382,29 @@ def search(self, query: str, limit: int = 5, filters: Optional[Dict[str, Any]] =
             logger.error(f"Error getting embedding for Query: {query}")
             return []
 
-        results = self.client.search(
-            collection_name=self.collection,
-            query_vector=query_embedding,
-            with_vectors=True,
-            with_payload=True,
-            limit=limit,
-        )
+        if not self.use_hybrid_search:
+            results = self.client.query_points(
+                collection_name=self.collection,
+                query=query_embedding,
+                with_vectors=True,
+                with_payload=True,
+                limit=limit,
+            ).points
+        else:
+            sparse_embedding = next(self.sparse_encoder.embed([query])).as_object()
```

Comment:
This whole section should be a new helper function, rather two functions- `hybrid_search()` and `async_hybrid_search()` using `client` and `async_client` respectively.